### DISCUSSION TITLE
__nuxt data__ and Is the static page spa (generated by SSG requesting server data for the first time), is it a duplicate request for data?

### DISCUSSION THREAD

**Discussion by gllinzbk1** [nuxt/nuxt#23197] [General]
__nuxt data__ and Is the static page spa (generated by SSG requesting server data for the first time), is it a duplicate request for data?

My current situation is SSG, which generates HTML and also requests server-side data
```
    const { data:data } = await useFetch('http://f.lgl.lb.in:8080/vueapi/list', {
  method: 'GET',
  query: {
    url: routeResult[0],
    page: 1
  },
  server: true,
  refetch:false,
  retry: 3,
  responseType:'json',
  watch:routeResult[0]
})
```

**Comment by gllinzbk1** [comment]
I cannot understand the generated static page, it already has nuxt_data Why do I need to access the server for the first time when accessing a static page

**Reply by gllinzbk1** [reply]
![image](https://github.com/nuxt/nuxt/assets/115514156/2422622f-b1cf-4d4c-aa46-aded3281bfad)

**Comment by noahps** [comment]
I'm not sure, but I feel like in some ways you are experiencing the same challenge I am [trying to solve here](https://github.com/nuxt/nuxt/discussions/23236#discussion-5631644). I'm not sure if my workaround using a combination of `useNuxtData` and `useFetch` might work for you? But I'm hoping more experienced people may be able to shed some light on how best to handle api data for static pages!

**Comment by fouss** [comment]
any update on this ? I also don't understand why on full static there is the payload.json + the injected script __NUXT_DATA__, this make my generated pages like 10 times bigger than the same with nuxt2
